src/aws/flb_aws_error_reporter.c

```diff
@@ -0,0 +1,243 @@
+/* -*- Mode: C; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*  Fluent Bit
+ *  ==========
+ *  Copyright (C) 2019-2021 The Fluent Bit Authors
+ *  Copyright (C) 2015-2018 Treasure Data Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+#include <monkey/mk_core/mk_list.h>
+
+#include <fluent-bit/flb_mem.h>
+#include <fluent-bit/flb_log.h>
+#include <fluent-bit/aws/flb_aws_error_reporter.h>
+
+/* helper function to get int type environment variable*/
+static int getenv_int(const char *name) {
+    char *value, *end;
+    long result;
+
+    value = getenv(name);
+    if (!value) {
+        return 0;
+    }
+
+    result = strtol(value, &end, 10);
+    if (*end != '\0') {
+        return 0;
+    }
+    return (int) result;
+}
+
+/* create an error reporter*/
+struct flb_aws_error_reporter *flb_aws_error_reporter_create()
+{
+    char *path_var = NULL;
+    int ttl_var, status_message_length;
+    struct flb_aws_error_reporter *error_reporter;
+
+    error_reporter = flb_calloc(1, sizeof(struct flb_aws_error_reporter));
+    if (!error_reporter) {
+        flb_errno();
+        return NULL;
+    }
+
+    /* setup error report file path */
+    path_var = getenv(STATUS_MESSAGE_FILE_PATH_ENV);
```

Comment:
instead of getenv, please use the flb_env.c family functions. so you create a context for your own purposes, the advantage is that users can define variables in the configuration file and that interface handles that + getenv if variable is not found.

Reply:
So it's kind of not working for this specific feature to use flb_env.c family function. The env setup inside this feature is from a dependency of AWS component, we need to use that env as a condition to make sure if this feature is enable or not. And only if we get that env, we just start initialize our customized logger. So it's not working to put a flb_env struct into logger because we need this env at very first and before our logger logic, thus we do not have a place to save the flb_env.